packages/legend-application-query/src/stores/QueryBuilderLambdaProcessor.ts

```diff
@@ -1186,6 +1186,7 @@ export class QueryBuilderLambdaProcessor
         valueSpecification,
         false,
       );
+      columnState.setNodeIdsForProjectionColumn();
```

Comment:
So the approach you're adopting is to find all the places where we create `projection column` then decorate the state with an `ID`, which you can use when rendering the explorer tree to check if the property is already used or not.

It's a sound approach. However, I think we should modify this slightly. With the current way, we decorate the projection column actively, which is fragile since we could miss spots. To me, for this feature, it's more an analytics than an attribute. As such, let's create a getter/computation instead.

For example, have a computation to get used properties in `QueryBuilderProjectionState`

```ts
get usedProperties(): AbstractProperty[]
```

or something like that. Then have explorer use this computation/analytics to render the tree.

Reply:
or maybe we can think of getting the ID with something like `get usedPropertyIds(): string[]`.

Regardless, I want the relationship between explorer (Exp) and fetch-structure (FS) states go like this:

```
Exp: Hey, can you tell me what you have used?
FS: Sure, this is the list ...
Exp: Thanks, let me try to understand it
```

At least, that's what I think makes sense for now. _I'm worried a bit that this kind of check could tangle up and make it harder for us later to do generic query builder, but maybe I'm just overthinking here_